packages/astro/src/default/layouts/Layout.astro

```diff
@@ -49,6 +49,9 @@ const { title } = Astro.props;
 
           if (newMain && oldMain) {
             builtInSwap.swapBodyElement(newMain, oldMain);
+
+            // delete extraneous route announcer
+            document.querySelector('.astro-route-announcer')?.remove();
```

Comment:
Unrelated fix to this PR but I noticed that as you navigate between pages we were accumulating `div`s under the body.